File: app/src/main/java/io/github/qwqgong/androidcyaml/runtime/RuntimeOverrideSettings.java
package io.github.qwqgong.androidcyaml;

record RuntimeOverrideSettings(
        ProcessMatchingMode processMatchingMode,
        boolean ipv6Enabled,
        RuntimeLogLevel logLevel,
        boolean adaptiveTcpConcurrent,
        boolean webViewXhttp,
        boolean lanWebUiPublic
) {
    RuntimeOverrideSettings {
        processMatchingMode = processMatchingMode == null
                ? ProcessMatchingMode.ALWAYS
                : processMatchingMode;
        logLevel = logLevel == null ? RuntimeLogLevel.WARNING : logLevel;
    }

    static RuntimeOverrideSettings defaults() {
        return new RuntimeOverrideSettings(
                ProcessMatchingMode.ALWAYS,
                true,
                RuntimeLogLevel.WARNING,
                false,
                false,
                false
        );
    }
}

File: app/src/main/java/io/github/qwqgong/androidcyaml/runtime/RuntimeOverrideSettings.kt
package io.github.qwqgong.androidcyaml

data class RuntimeOverrideSettings(
    val processMatchingMode: ProcessMatchingMode,
    val ipv6Enabled: Boolean,
    val logLevel: RuntimeLogLevel,
    val adaptiveTcpConcurrent: Boolean,
    val webViewXhttp: Boolean,
    val lanWebUiPublic: Boolean,
) {
    companion object {
        fun defaults(): RuntimeOverrideSettings = RuntimeOverrideSettings(
            ProcessMatchingMode.ALWAYS,
            true,
            RuntimeLogLevel.WARNING,
            false,
            false,
            false,
        )
    }
}
